src/main/frontend/src/components/calendar/calendarUtils.js

```diff
@@ -63,7 +63,7 @@ const checkIfPublicHoliday = (day, publicHolidays, personalData) => {
 const checkIfUserDay = dayObject => {
   const { day, publicHoliday, personalData } = dayObject;
 
-  if (!personalData.length) {
+  if (personalData !== undefined && !personalData.length) {
```

Comment:
```suggestion
  if (personalData != null && !personalData.length) {
```

This checks for both `null` and `undefined`. 💃